conda_build/build.py

```diff
@@ -2750,6 +2756,224 @@ def write_test_scripts(metadata, env_vars, py_files, pl_files, lua_files, r_file
                            lua_files, r_files, shell_files, trace)
     return test_run_script, test_env_script
 
+def determine_ext_and_win_check():
+    win_check = False
+    if sys.platform == "darwin":
+        ext = ".dylib"
+    elif sys.platform == "win32":
+        ext = ".dll"
+        win_check = True
+    else:
+        ext = ".so"
+    return ext, win_check
```

Comment:
This should inferred from the package's target_platform instead of where conda-build is run.